src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,513 @@
+            throw new NoDescriptionException();
+        } else if (!command.startsWith("echo ")) {
+            throw new NoSpaceException();
+        }
+    }
+
+    public static void handleEcho(String command) {
+        boolean proceed = true;
+        System.out.println(SEPARATE_LINE);
+        try {
+            checkEcho(command);
+        } catch (NoDescriptionException e) {
+            System.out.println("Sorry, I have no idea what you want me to echo");
+            proceed = false;
+        } catch (NoSpaceException e) {
+            System.out.println("Please use a space to separate command, task, and time (if any)");
+            proceed = false;
+        }
+
+        if (proceed) {
+            System.out.println(command.substring(5));
+        }
+        System.out.println(SEPARATE_LINE);
+    }
+
+    public static void checkToDo(String command) throws NoDescriptionException, NoSpaceException{
+        if (command.equals("todo") || command.equals("todo ")) {
+            throw new NoDescriptionException();
+        } else if (!command.startsWith("todo ")) {
+            throw new NoSpaceException();
+        }
+    }
+
+    public static boolean proceedTodo(String command) {
+        try {
+            checkToDo(command);
+        } catch (NoDescriptionException e) {
+            System.out.println("What task do you want to add?");
+            return false;
+        } catch (NoSpaceException e) {
+            System.out.println("Please use a space to separate command, task, and time (if any)");
+            return false;
+        }
+        return true;
+    }
+
+    public static void printToDo(Task task, int size) {
+        System.out.println("Got it. I've added this task: \n" + "\t" + task.toString());
+        System.out.println("Now you have " + size + " tasks in the list.");
+    }
+
+    public static void checkDeadline(String command) throws NoDescriptionException, NoTimeException, NoSpaceException, WrongTimeTypeException{
+        int slashIndex = command.indexOf("/");
+        if (command.equals("deadline") || command.equals("deadline ")) {
+            throw new NoDescriptionException();
+        } else if (!command.startsWith("deadline ")) {
+            throw new NoSpaceException();
+        } else if (!command.contains("/")) {
+            throw new NoTimeException();
+        }  else if (slashIndex == 9) {
+            throw new NoDescriptionException();
+        }  else if (command.charAt(slashIndex - 1) != ' ') {
+            throw new NoSpaceException();
+        } else if ((command.length() == slashIndex + 3)) {
+            throw new NoTimeException();
+        } else if ((command.length() == (slashIndex + 4)) && (command.charAt(slashIndex + 3) == ' ')) {
+            throw new NoTimeException();
+        } else if (command.charAt(slashIndex + 3) != ' ') {
+            throw new NoSpaceException();
+        } else if (command.contains("/at")) {
+            throw new WrongTimeTypeException();
+        }
```

Comment:
may be can convert this chunk of code to the correspond switch case block for ease of reading